llvm/lib/Target/ARM/ARMSelectionDAGInfo.cpp

```diff
@@ -192,6 +302,12 @@ SDValue ARMSelectionDAGInfo::EmitTargetCodeForMemcpy(
     return EmitSpecializedLibcall(DAG, dl, Chain, Dst, Src, Size,
                                   Alignment.value(), RTLIB::MEMCPY);
 
+  if (Subtarget.isCortexM7() &&
```

Comment:
Done. I've opted by just using a boolean check for terseness. Do let me know if you'd prefer an ST feature

Reply:
> I've opted by just using a boolean check for terseness. Do let me know if you'd prefer an ST feature
I see that this causes failures since it triggers the optimisation for other cores. So I'll go for your original suggestion

Reply:
Done